tests/micro/arduino/test_arduino_workflow.py

```diff
@@ -49,6 +51,30 @@ def project_dir(workspace_dir):
     return workspace_dir / "project"
 
 
+# Saves the Arduino project's state, runs the test, then resets it
+@pytest.fixture(scope="function")
+def does_not_affect_state(project_dir):
```

Comment:
maybe name this save_generated_project_state? also, i think this should depend on `project` fixture, even if it doesn't use it directly in the function?

Reply:
I agree `save_generated_project_state` and depending on the `project` fixture is clearer. At some point, I might want to break these tests out into their own file, but for now I'll probably keep it as is.

Reply:
Fixed, as `save_generated_project_state` has been removed and the relevant tests have been broken out into their own file.